source/config/profileUpgradeSteps.py

```diff
@@ -439,6 +440,31 @@ def upgradeConfigFrom_13_to_14(profile: ConfigObj):
 	log.debug("Deleting old config value.")
 	del profile["speech"]["outputDevice"]
 
+	# """Convert keyboard typing echo configuration from boolean to integer values."""
```

Comment:
You should rather do your own profile upgrade IMO, i.e. 14 to 15. Indeed, in the following cases, the upgrade of typing echo will not be done:
* If the user is on alpha channel and has upgraded each day, i.e. has already executed profile upgrade from 13 to 14 but without the typing echo upgrade code
* overall, even upgrading from 2024.4.1 to 2025.1 (stable channel), if no output device is present in config, because the function `upgradeConfigFrom_13_to_14` will return early without evaluating typing eco code.